pkg/common/volume_operations.go

```diff
@@ -424,6 +428,78 @@ func (vo *VolumeOperationsImpl) WaitStatus(ctx context.Context, volumeID string,
 	}
 }
 
+// ExpandVolume updates Volume status to Resizing to trigger expansion in reconcile, if volume has already had status
+// Resizing or Resized, function doesn't do anything. In case of statuses beside VolumeReady, Created, Published function return error
+// Receive golang context, volume CR, requiredBytes as int
+// Return volume spec, error
+func (vo *VolumeOperationsImpl) ExpandVolume(ctx context.Context, volume *volumecrd.Volume, requiredBytes int64) (*api.Volume, error) {
+	defer vo.metrics.EvaluateDurationForMethod("ExpandVolume")()
+	ll := vo.log.WithFields(logrus.Fields{
+		"method":   "ExpandVolume",
+		"volumeID": volume.Spec.Id,
+	})
+	currStatus := volume.Spec.CSIStatus
+	switch currStatus {
+	case apiV1.Resizing, apiV1.Resized:
+		ll.Debug("Volume is already expanding")
+	case apiV1.VolumeReady, apiV1.Created, apiV1.Published:
+		if !util.IsStorageClassLVG(volume.Spec.StorageClass) {
+			return nil, status.Error(codes.FailedPrecondition, "Volume doesn't have LVG storage class")
```

Comment:
Please rename `Storage class <name> doesn't support volume expansion`. Does it make sense to check corresponding field in SC? User might create new LVG SC and disable expansion.

Reply:
If SC doesn't support resizing, it doesn't have field `allowVolumeExpansion: true` in spec. In this case, csi-resizer doesn't call ControllerExpandVolume for unsupported SC. So this check is redundant because if SC allows volume expansion, it has a corresponding field in spec, and resizer call request only for such SC. But if User creates not LVG SC and enables expansion, it makes sense to check if volume SC is LVG.

Reply:
Ok, got it!